server/audit_math/suite.py

```diff
@@ -0,0 +1,660 @@
+# pylint: disable=invalid-name
+"""
+This library allows the performing of a stratified audit. For now, it works with
+ballot polling and ballot comparison, plurality winner contests. The method
+is SUITE, described by Ottoboni, Stark, Lindeman, and McBurnett here:
+https://arxiv.org/abs/1809.04235
+
+This code borrows heavily from code already written by Stark and Ottoboni here:
+https://github.com/pbstark/CORLA18
+
+
+"""
+from itertools import product
+import math
+from typing import Tuple, Dict
+
+# from decimal import Decimal
+import numpy as np
+import scipy as sp
+
+
+from .sampler_contest import Contest, CVRS
+from . import bravo
+
+
+GAMMA = 1.03905  # This GAMMA is used in Stark's tool, AGI, and CORLA
+MIN_SAMPLE_SIZE = 5  # The smallest sample size we want to take
+
+
+class BallotPollingStratum:
+    """
+    A class encapsulating a stratum of ballots in an election. Each stratum is its
+    own contest object, with its own margin. Strata, along with the overall
+    contest object, are passed to the SUITE module when perfoming mixed-strata
+    audits.
+    """
+
+    SAMPLE_RESULTS = Dict[str, Dict[str, int]]  # ballot polling
+
+    num_ballots: int
+    vote_totals: Dict[str, int]
+    sample: SAMPLE_RESULTS
+    sample_size: int
+
+    def __init__(
+        self,
+        num_ballots: int,
+        vote_totals: Dict[str, int],
+        sample_results: SAMPLE_RESULTS,
+        sample_size: int,
+    ):
+
+        """
+        initialize this stratum.
+
+        Inputs:
+            num_ballots: the total number of ballots cast in this stratum
+            vote_totals: the per-candidate vote totals for this stratum, of the form
+                    {
+                        cand1: 700,
+                        cand2: 200,
+                        ...
+                    }
+            sample_results : the vote totals for this stratum in the sample so far. E.g.,
+                    {
+                        "winner1": 10,
+                        "winner2": 7,
+                        "loser1": 5,
+                        "loser2": 2,
+                        ...
+                    }
```

Comment:
i think sample_results is a dict mapping round id to the vote totals example you have here

Reply:
Oh you're right, will fix.